### PR TITLE
Implement support for SMF server replacement compatibility

### PR DESCRIPTION
**Pull request by atampy25**
This is currently just so that Cirrus will build it for me, since Peacock fails to build with a series of bizarre errors whenever I try

### CODE REVIEW

**Comment by atampy25** [comment]
LGTM

File: components/controller.ts [hunk 1]
```diff
@@ -479,6 +479,65 @@ export class Controller {
...
+                parse(readFileSync(modFrameworkDataPath!).toString()) as {
+                    lastServerSideStates: {
+                        unlockables?: Unlockable[]
+                        contracts?: {
+                            [k: string]: MissionManifest
+                        }
+                        blobs?: Record<string, string>
+                    }
+                }
+            ).lastServerSideStates
```
Comment (AnthonyFuller) [inline components/controller.ts:498]: You might as well type this properly, a type may already exist for it as we read it for the deployed mods. If it does not, it should be refactored to use a type.

**Review by AnthonyFuller — CHANGES_REQUESTED** [review]
Other than the comment, it LGTM.

File: components/controller.ts (hunk 1 above)
Comment (AnthonyFuller) [reply components/controller.ts:498]: Then see the latter part of my comment.

**Review by AnthonyFuller — APPROVED** [review]
LGTM

**Pull request merged**